core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -2011,12 +2012,68 @@ class KafkaConfig private(doLog: Boolean, val props: java.util.Map[_, _], dynami
       "offsets.commit.required.acks must be greater or equal -1 and less or equal to offsets.topic.replication.factor")
     require(BrokerCompressionCodec.isValid(compressionType), "compression.type : " + compressionType + " is not valid." +
       " Valid options are " + BrokerCompressionCodec.brokerCompressionOptions.mkString(","))
-    require(!processRoles.contains(ControllerRole) || controllerListeners.nonEmpty,
-      s"${KafkaConfig.ControllerListenerNamesProp} cannot be empty if the server has the controller role")
-
     val advertisedListenerNames = advertisedListeners.map(_.listenerName).toSet
+    if (usesSelfManagedQuorum) {
+      require(controlPlaneListenerName.isEmpty,
+        s"${KafkaConfig.ControlPlaneListenerNameProp} is not supported in KRaft mode.")
```

Comment:
Used to be checked in `BrokerServer` -- now we just do it here

Reply:
It looks like we have section for validating this property. Maybe move this error there:
```scala
      // validate control.plane.listener.name config
      if (controlPlaneListenerName.isDefined) {
        require(advertisedListenerNames.contains(controlPlaneListenerName.get),
          s"${KafkaConfig.ControlPlaneListenerNameProp} must be a listener name defined in ${KafkaConfig.AdvertisedListenersProp}. " +
          s"The valid options based on currently configured listeners are ${advertisedListenerNames.map(_.value).mkString(",")}")
        // controlPlaneListenerName should be different from interBrokerListenerName
        require(!controlPlaneListenerName.get.value().equals(interBrokerListenerName.value()),
          s"${KafkaConfig.ControlPlaneListenerNameProp}, when defined, should have a different value from the inter broker listener name. " +
          s"Currently they both have the value ${controlPlaneListenerName.get}")
      }
```